### PR TITLE
fix: Handle more validation errors as field errors

### PR DESCRIPTION
**Pull request by karlkfi**
- Using field.Error allows more errors to be wrapped as a
  ValidationError, instead of interrupting validation and exiting
  early.
- Add explicit validation for kind (clearer error).
- Move NestedField from testutil to object, so it can be used at
  runtime.

### CODE REVIEW

**Review by mortent — APPROVED** [review]
/approve

**Comment by karlkfi** [comment]
@ash2k can you take a look at this? It's extracted from https://github.com/kubernetes-sigs/cli-utils/pull/505 which you already approved.

File: pkg/object/field.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		Type:     field.ErrorTypeNotFound,
+		Field:    FieldPath(fieldPath),
+		BadValue: value,
+		Detail:   "",
+	}
+}
+
+// Simplistic jsonpath formatter, just for NestedField errors.
+func FieldPath(fieldPath []interface{}) string {
+	path := ""
```
Comment (ash2k) [inline pkg/object/field.go:81]: nit: could use `strings.Builder` here for concatenation.

**Review by ash2k — APPROVED** [review]
/lgtm

File: pkg/object/field.go (hunk 1 above)
Comment (karlkfi) [reply pkg/object/field.go:81]: good catch: https://github.com/kubernetes-sigs/cli-utils/pull/507

**Pull request merged**